src/components/QueryEditor/index.test.tsx

```diff
@@ -37,22 +39,127 @@ describe('QueryEditorForm', () => {
+  describe('Alias field', () => {
+    it('Should correctly render and trigger changes on blur', () => {
+      const alias = '{{metric}}';
+      const query: OpenSearchQuery = {
+        refId: 'A',
+        query: '',
+        alias,
+        metrics: [
+          {
+            id: '1',
+            type: 'count',
+          },
+        ],
+        bucketAggs: [
+          {
+            type: 'date_histogram',
+            id: '2',
+          },
+        ],
+      };
+
+      render(
+        <QueryEditor query={query} onChange={mockOnChange} onRunQuery={mockRunQuery} datasource={mockDatasource} />
+      );
+
+      let aliasField = screen.getByLabelText('Alias') as HTMLInputElement;
+
+      // The Query should have an alias field
+      expect(aliasField).toBeInTheDocument();
+
+      // its value should match the one in the query
+      expect(aliasField.value).toBe(alias);
+
+      // We change value and trigger a blur event to trigger an update
+      const newAlias = 'new alias';
+      fireEvent.change(aliasField, { target: { value: newAlias } });
+      fireEvent.blur(aliasField);
+
+      // the onChange handler should have been called correctly, and the resulting
+      // query state should match what expected
+      expect(mockOnChange).toHaveBeenCalledTimes(1);
+      expect(mockOnChange.mock.calls[0][0].alias).toBe(newAlias);
+    });
+    it('Should not be shown if query is Lucene Traces', () => {
+      const query: OpenSearchQuery = {
+        refId: 'A',
+        query: '',
+        luceneQueryType: LuceneQueryType.Traces,
+        metrics: [
+          {
+            id: '1',
+            type: 'avg',
+          },
+        ],
+        bucketAggs: [{ id: '2', type: 'terms' }],
+      };
+
+      render(
+        <QueryEditor query={query} datasource={mockDatasource} onChange={mockOnChange} onRunQuery={mockRunQuery} />
+      );
+
+      expect(screen.queryByLabelText('Alias')).toBeNull();
+    });
+    it('Should not be shown if query is PPL', () => {
+      const query: OpenSearchQuery = {
+        refId: 'A',
+        query: '',
+        queryType: QueryType.PPL,
+      };
+
+      render(
+        <QueryEditor query={query} datasource={mockDatasource} onChange={mockOnChange} onRunQuery={mockRunQuery} />
+      );
+
+      expect(screen.queryByLabelText('Alias')).toBeNull();
+    });
+
+    it('Should not be shown if last bucket aggregation is not Date Histogram', () => {
+      const query: OpenSearchQuery = {
+        refId: 'A',
+        query: '',
+        metrics: [
+          {
+            id: '1',
+            type: 'avg',
+          },
+        ],
+        bucketAggs: [{ id: '2', type: 'terms' }],
+      };
+
+      render(
+        <QueryEditor query={query} datasource={mockDatasource} onChange={mockOnChange} onRunQuery={mockRunQuery} />
+      );
+
+      expect(screen.queryByLabelText('Alias')).toBeNull();
+    });
+
+    it('Should be shown if last bucket aggregation is Date Histogram', () => {
```

Comment:
Should this test have more than 1 bucket arg?

Reply:
Good catch, fixed!